### PR TITLE
Add a workload that randomly generates different transaction shapes

### PR DESCRIPTION
**Pull request by halfprice**
## Description 

Randomized transaction testing.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

**Review by aschran — COMMENTED** [review]
looks great overall in terms of the PR itself! i am curious what our longer term story is for this framework vs. sui surfer, do we want to keep both of them or move towards combining? curious if you or Mark have opinions

File: crates/sui-benchmark/tests/simtest.rs [hunk 1]
```diff
@@ -558,7 +558,7 @@ mod test {
         let mut simulated_load_config = SimulatedLoadConfig::default();
         {
             let mut rng = thread_rng();
-            simulated_load_config.shared_counter_weight = if rng.gen_bool(0.5) { 5 } else { 50 };
+            simulated_load_config.shared_counter_weight = if rng.gen_bool(0.5) { 5 } else { 5 };
```
[the remark this answers is not in the document]
Comment (halfprice) [reply crates/sui-benchmark/tests/simtest.rs]: Oops. Changed back.

File: crates/sui-benchmark/tests/simtest.rs [hunk 2]
```diff
@@ -955,6 +956,7 @@ mod test {
                 shared_deletion_weight: 1,
                 shared_counter_hotness_factor: 50,
                 randomness_weight: 1,
+                randomized_transaction_weight: 1,
```
[the remark this answers is not in the document]
Comment (halfprice) [reply crates/sui-benchmark/tests/simtest.rs:959]: It's somewhere around 40~50 transactions in one run of test_simulated_load_shared_object_congestion_control

**Comment by halfprice** [comment]
Thanks @aschran for the review!

**Review by arun-koshy — APPROVED** [review]
Very cool! Thanks for adding this workload

**Pull request merged**